### DISCUSSION TITLE
Where is pigpio and pigpiod located in the file system?

### DISCUSSION THREAD

**Discussion by paulvee** [Poeschl/Hassio-Addons#301] [Q&A] [answered]
I looked just about everywhere but I can't find the location of pigpiod in the HA structure.

Using Pyscript, I can import pigpio and use it, but I would like to 
Here is my sys.path as reported by Python:
/config/deps/lib/python3.9/site-packages', '/config', '/usr/local/lib/python39.zip', '/usr/local/lib/python3.9', '/usr/local/lib/python3.9/lib-dynload', '/usr/local/lib/python3.9/site-packages', '/usr/src/homeassistant']

Does anybody know or can help?

**Comment by Poeschl** [comment, marked as answer]
Pigpiod is not available in the HA structure. The Pigpio integration requires your setup to have an usable pigpio on your system which can be accessed via sockets. This will break after 2022.04.

So one workaround on a supervised installation or on Home Assistant OS is to use the [remote_rpi_pgio integration](https://www.home-assistant.io/integrations/remote_rpi_gpio/) and connect to the Pigpio addon in this repository. Inside of the addon the pigpiod is running and accepting commands via http.

**Reply by paulvee** [reply]
Markus, I tried pigpio 1.5.0 with the -s 1 option and that works fine. Thank you!
I'm puzzled with your message that pigpio will break as well as RPi.gpio after 2022.04. I was under the impression that because they are two different and separate implementations, pigpio could survive. BTW, I'm using HACS pyscript right now and that works really well because you can freely import modules.

**Reply by Poeschl** [reply]
All integrations which access the gpios locally will be removed on 2022.04. As far as I know only the remote_rpi_gpio integration will stay in the homeassistant core, since it communicates via rest calls. It can still be used to connect to "localhost" where the addon will respond and control the gpio. 

Any integration over HACS will continue to work as well.

**Reply by paulvee** [reply]
Yeah, after carefully re-reading the information I'm afraid you're right. We'll see what the HACS folks can or will be able to do. I'm not too fond of the remote_rpi_gpio myself, but time will tell if that's the only remaining method unless you use a second RPi or an ESP to control hardware via MQTT. Seems overly complicated...

**Reply by Poeschl** [reply]
Technically the core integration could be copied as Community integration via HACS. But it needs to be maintained to stay up to date.
But time will tell.... 😶
